src/Magento/FunctionalTestingFramework/Console/RunTestFailedCommand.php

```diff
@@ -0,0 +1,194 @@
+
+use Magento\FunctionalTestingFramework\Config\MftfApplicationConfig;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+
+class RunTestFailedCommand extends BaseGenerateCommand
+{
+    /**
+     * Default Test group to signify not in suite
+     */
+    const DEFAULT_TEST_GROUP = 'default';
+
+    const TESTS_OUTPUT_DIR = TESTS_BP .
+    DIRECTORY_SEPARATOR .
+    "tests" .
+    DIRECTORY_SEPARATOR .
+    "_output" .
+    DIRECTORY_SEPARATOR;
+
+    const TESTS_FAILED_FILE = self::TESTS_OUTPUT_DIR . "failed";
+    const TESTS_RERUN_FILE = self::TESTS_OUTPUT_DIR . "rerun_tests";
+    const TESTS_MANIFEST_FILE= TESTS_MODULE_PATH .
+    DIRECTORY_SEPARATOR .
+    "_generated" .
+    DIRECTORY_SEPARATOR .
+    "testManifest.txt";
+
+    /**
+     * @var array
+     */
+    private $failedList = [];
+
+    /**
+     * Configures the current command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this->setName('run:failed')
+            ->setDescription('Execute a set of tests referenced via failed file');
+
+        parent::configure();
+    }
+
+    /**
+     * Executes the current command.
+     *
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     * @return integer|null|void
+     * @throws \Exception
+     *
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        // Create Mftf Configuration
+        MftfApplicationConfig::create(
+            false,
+            MftfApplicationConfig::GENERATION_PHASE,
+            false,
+            false
+        );
+
+        $testConfiguration = $this->getFailedTestList();
+
+        if ($testConfiguration === null) {
+            return null;
+        }
+
+        $command = $this->getApplication()->find('generate:tests');
+        $args = ['--tests' => $testConfiguration, '--remove' => true];
+
+        $command->run(new ArrayInput($args), $output);
+
+        $testManifestList = $this->readTestManifestFile();
+
+        foreach ($testManifestList as $testCommand) {
+            $codeceptionCommand = realpath(PROJECT_ROOT . '/vendor/bin/codecept') . ' run functional ';
+            $codeceptionCommand .= $testCommand;
+
+            $process = new Process($codeceptionCommand);
+            $process->setWorkingDirectory(TESTS_BP);
+            $process->setIdleTimeout(600);
+            $process->setTimeout(0);
+            $process->run(
+                function ($type, $buffer) use ($output) {
+                    $output->write($buffer);
+                }
+            );
+            if (realpath(self::TESTS_FAILED_FILE)) {
```

Comment:
For some reason, this `realpath` still allows the conditional to pass if the fiel has been deleted.

Locally, I've found that adding `file_get_contents()` to the `readFailedTestFile` function works to prevent the nulling out of `$this->failedTests`:
```private function readFailedTestFile($filePath)
{
    if (file_get_contents($filePath) !== false) {
        return file($filePath, FILE_IGNORE_NEW_LINES);
    }
    return [];
}```